docs/core-concepts/index.md

```diff
@@ -0,0 +1,32 @@
+# Core Concepts
+
+These are some of the core concepts that you should be familiar with before going over the [Getting Started Guide](get-started/index.md).
+
+
+### **Git Source**
+The git source represents a git repository in Flux.
+
+For more information, take a look at [this documentation](components/source/gitrepositories.md)
+
+### **Kustomization**
+The kustomization represents an application that flux is supposed to reconcile in the cluster. It specifies a path in the git repository containing the `kustomization.yaml` for the application. If the repository contains only plain kubernetes manifests, a `kustomization.yaml` is generated for you. If there are non-kubernetes related files or manifest that you don't want to apply to the cluster, you can exclude it under the `spec.ignore` fields.
+
+For more information, take a look at [this documentation](components/kustomize/kustomization.md)
+
+### **Reconciliation**
+Reconcilation refers to ensuring the state(different resources e.g deployments) of a particular application(represented by a kustomization) in the cluster matches the state defined in a git repository. It performs this reconciliation every thirty seconds by default but this can be specified in the kustomization. If you make any changes to the cluster using `kubectl edit` or `kubectl patch`, it will be promptly reverted. You either suspend the reconciliation or push your changes to a git repository.
```

Comment:
The default reconciliation time is 1m when generating the Kustomization with `flux` CLI.